### ISSUE TITLE
Crash on killing application

### ISSUE THREAD

**Issue by Hwan-seok** [AgoraIO-Extensions/Agora-Flutter-SDK#700]
**Describe the bug**
```
Fatal Exception: NSInternalInconsistencyException
0  CoreFoundation                 0x99288 __exceptionPreprocess
1  libobjc.A.dylib                0x16744 objc_exception_throw
2  Foundation                     0x132360 _userInfoForFileAndLine
3  Flutter                        0x1256c (누락 UUID bab62788d88339b192aed18e13dece71)
4  agora_rtc_engine               0xbdc4 EventHandler::OnEvent(char const*, char const*) + 20 (FlutterIrisEventHandler.mm:20)
5  AgoraRtcWrapper                0x62bdc agora::iris::rtc::RtcEngineEventHandler::onApiCallExecuted(int, char const*, char const*) + 1423 (string:1423)
6  AgoraRtcKit                    0x36fc18 (누락 UUID 255c10086c353680b0bdbcaa02df8cdc)
7  AgoraRtcKit                    0x36f818 (누락 UUID 255c10086c353680b0bdbcaa02df8cdc)
8  AgoraRtcKit                    0x307fc8 (누락 UUID 255c10086c353680b0bdbcaa02df8cdc)
9  AgoraRtcKit                    0x307f38 (누락 UUID 255c10086c353680b0bdbcaa02df8cdc)
10 AgoraCore                      0xcec4 (누락 UUID 0fb485d0470236bf8c0a893f74d704f4)
11 AgoraCore                      0xe014 (누락 UUID 0fb485d0470236bf8c0a893f74d704f4)
12 AgoraCore                      0xdc40 (누락 UUID 0fb485d0470236bf8c0a893f74d704f4)
13 AgoraCore                      0xd218 (누락 UUID 0fb485d0470236bf8c0a893f74d704f4)
14 AgoraCore                      0x18408 (누락 UUID 0fb485d0470236bf8c0a893f74d704f4)
15 libsystem_pthread.dylib        0x19ac _pthread_start
16 libsystem_pthread.dylib        0xe68 thread_start
```
```
Crashed: AgoraEvent
0  FirebaseCrashlytics            0x223ac FIRCLSProcessRecordAllThreads + 393 (FIRCLSProcess.c:393)
1  FirebaseCrashlytics            0x2278c FIRCLSProcessRecordAllThreads + 424 (FIRCLSProcess.c:424)
2  FirebaseCrashlytics            0x187b4 FIRCLSHandler + 34 (FIRCLSHandler.m:34)
3  FirebaseCrashlytics            0x129b0 __FIRCLSExceptionRecord_block_invoke + 232 (FIRCLSException.mm:232)
4  libdispatch.dylib              0x3a30 _dispatch_client_callout + 20
5  libdispatch.dylib              0x12c90 _dispatch_lane_barrier_sync_invoke_and_complete + 56
6  FirebaseCrashlytics            0x11790 FIRCLSExceptionRecord + 234 (FIRCLSException.mm:234)
7  FirebaseCrashlytics            0x12458 FIRCLSExceptionRecordNSException + 126 (FIRCLSException.mm:126)
8  FirebaseCrashlytics            0x11390 FIRCLSTerminateHandler() + 398 (FIRCLSException.mm:398)
9  libc++abi.dylib                0x10f18 std::__terminate(void (*)()) + 20
10 libc++abi.dylib                0x13c1c __cxa_get_exception_ptr + 34
11 libc++abi.dylib                0x13bc8 __cxxabiv1::exception_cleanup_func(_Unwind_Reason_Code, _Unwind_Exception*) + 138
12 libobjc.A.dylib                0x168ac objc_exception_throw + 420
13 Foundation                     0x132360 _userInfoForFileAndLine + 202
14 Flutter                        0x1256c (누락 UUID bab62788d88339b192aed18e13dece71)
15 agora_rtc_engine               0xbdc4 EventHandler::OnEvent(char const*, char const*) + 20 (FlutterIrisEventHandler.mm:20)
16 AgoraRtcWrapper                0x62bdc agora::iris::rtc::RtcEngineEventHandler::onApiCallExecuted(int, char const*, char const*) + 1423 (string:1423)
17 AgoraRtcKit                    0x36fc18 agora::rtc::RtcEngineParameters::setInEarMonitoringVolume(int) + 50116
18 AgoraRtcKit                    0x36f818 agora::rtc::RtcEngineParameters::setInEarMonitoringVolume(int) + 49092
19 AgoraRtcKit                    0x307fc8 (누락 UUID 255c10086c353680b0bdbcaa02df8cdc)
20 AgoraRtcKit                    0x307f38 (누락 UUID 255c10086c353680b0bdbcaa02df8cdc)
21 AgoraCore                      0xcec4 ahpl_perf_set_callback + 1480
22 AgoraCore                      0xe014 ahpl_mpq_itc_ack + 252
23 AgoraCore                      0xdc40 ahpl_mpq_loop + 84
24 AgoraCore                      0xd218 ahpl_perf_set_callback + 2332
25 AgoraCore                      0x18408 ahpl_profile_clear_registered + 2488
26 libsystem_pthread.dylib        0x19ac _pthread_start + 148
27 libsystem_pthread.dylib        0xe68 thread_start + 8
```

**Smartphone (please complete the following information):**
 - Device: IOS

**Comment by littleGnAl** [comment]
@Hwan-seok Thanks for reporting, we will fix it on the next release.

**Comment by albaschez** [comment]
Happening with flutter 3.0.5 and versions 5.2 and 5.3. When using `joingCall` the app just crashes, no trace

**Comment by nizwar** [comment]
Still happened on 5.3.0, i'm not sure if it same issue but i got this from crashlytics
```
Fatal Exception: NSInternalInconsistencyException
0  CoreFoundation                 0x129dc0 __exceptionPreprocess
1  libobjc.A.dylib                0x287a8 objc_exception_throw
2  CoreFoundation                 0x2ba8c -[CFPrefsSearchListSource addManagedSourceForIdentifier:user:]
3  Foundation                     0x7e4c4 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:]
4  Flutter                        0x12d54 (Missing UUID c8164c661c033398832ad3cbdb9a3cd8)
5  agora_rtc_engine               0xb920 (Missing UUID 8c7c0a7cd37a30fa8b7b29d737abcacf)
6  AgoraRtcWrapper                0x68608 (Missing UUID 7c930d34a18733fd9128a97f17bf44b2)
7  AgoraRtcKit                    0x3d8df8 (Missing UUID 1ec6daa1d1f23e3dae9a2685ee010cd3)
8  AgoraRtcKit                    0x3d6fac (Missing UUID 1ec6daa1d1f23e3dae9a2685ee010cd3)
9  AgoraRtcKit                    0x36b42c (Missing UUID 1ec6daa1d1f23e3dae9a2685ee010cd3)
10 AgoraRtcKit                    0x36b39c (Missing UUID 1ec6daa1d1f23e3dae9a2685ee010cd3)
11 AgoraCore                      0xc9d0 (Missing UUID e9a448df109039e8a72e96e8aca0bd93)
12 AgoraCore                      0xdae0 (Missing UUID e9a448df109039e8a72e96e8aca0bd93)
13 AgoraCore                      0xd6b4 (Missing UUID e9a448df109039e8a72e96e8aca0bd93)
14 AgoraCore                      0xcc78 (Missing UUID e9a448df109039e8a72e96e8aca0bd93)
15 AgoraCore                      0x18358 (Missing UUID e9a448df109039e8a72e96e8aca0bd93)
16 libsystem_pthread.dylib        0x1bfc (Missing UUID 496dc4232dd43031bccf93e889035a34)
17 libsystem_pthread.dylib        0xa758 (Missing UUID 496dc4232dd43031bccf93e889035a34)
```

```
Crashed: AgoraEvent
0  Runner                         0x5ed68 FIRCLSProcessRecordAllThreads + 392 (FIRCLSProcess.c:392)
1  Runner                         0x5f148 FIRCLSProcessRecordAllThreads + 423 (FIRCLSProcess.c:423)
2  Runner                         0x55d20 FIRCLSHandler + 34 (FIRCLSHandler.m:34)
3  Runner                         0x51cb4 __FIRCLSExceptionRecord_block_invoke + 218 (FIRCLSException.mm:218)
4  libdispatch.dylib              0x481c _dispatch_client_callout + 20
5  libdispatch.dylib              0x13910 _dispatch_lane_barrier_sync_invoke_and_complete + 60
6  Runner                         0x51478 FIRCLSExceptionRecord + 225 (FIRCLSException.mm:225)
7  Runner                         0x5175c FIRCLSExceptionRecordNSException + 111 (FIRCLSException.mm:111)
8  Runner                         0x5105c FIRCLSTerminateHandler() + 279 (FIRCLSException.mm:279)
9  libc++abi.dylib                0x12fa0 std::__terminate(void (*)()) + 20
10 libc++abi.dylib                0x15c0c __cxa_get_exception_ptr + 34
11 libc++abi.dylib                0x15bb8 __cxxabiv1::exception_cleanup_func(_Unwind_Reason_Code, _Unwind_Exception*) + 138
12 libobjc.A.dylib                0x288d4 objc_exception_throw + 360
13 CoreFoundation                 0x2ba8c -[CFPrefsSearchListSource addManagedSourceForIdentifier:user:] + 98
14 Foundation                     0x7e4c4 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 132
15 Flutter                        0x12d54 (Missing UUID c8164c661c033398832ad3cbdb9a3cd8)
16 agora_rtc_engine               0xb920 EventHandler::OnEvent(char const*, char const*) + 304
17 AgoraRtcWrapper                0x68608 agora::iris::rtc::RtcEngineEventHandler::onAudioVolumeIndication(agora::rtc::AudioVolumeInfo const*, unsigned int, int) + 1412
18 AgoraRtcKit                    0x3d8df8 agora::rtc::RtcEngineParameters::setInEarMonitoringVolume(int) + 57384
19 AgoraRtcKit                    0x3d6fac agora::rtc::RtcEngineParameters::setInEarMonitoringVolume(int) + 49628
20 AgoraRtcKit                    0x36b42c (Missing UUID 1ec6daa1d1f23e3dae9a2685ee010cd3)
21 AgoraRtcKit                    0x36b39c (Missing UUID 1ec6daa1d1f23e3dae9a2685ee010cd3)
22 AgoraCore                      0xc9d0 ahpl_perf_set_callback + 1500
23 AgoraCore                      0xdae0 ahpl_mpq_itc_ack + 340
24 AgoraCore                      0xd6b4 ahpl_mpq_loop + 84
25 AgoraCore                      0xcc78 ahpl_perf_set_callback + 2180
26 AgoraCore                      0x18358 ahpl_profile_clear_registered + 2488
27 libsystem_pthread.dylib        0x1bfc _pthread_start + 320
28 libsystem_pthread.dylib        0xa758 thread_start + 8
```

**Comment by littleGnAl** [comment]
Is there any reason output for `NSInternalInconsistencyException`?

**Comment by Hwan-seok** [comment]
@littleGnAl Same issue with @nizwar came out since 5.3.0 and its portion is rapidly increasing.
My log has more information I think

```
Fatal Exception: NSInternalInconsistencyException
0  CoreFoundation                 0x99d78 __exceptionPreprocess
1  libobjc.A.dylib                0x16734 objc_exception_throw
2  Foundation                     0x1311f0 _userInfoForFileAndLine
3  Flutter                        0x12d54 (누락 UUID b5f419596300357c85f161f5e6e0920d)
4  agora_rtc_engine               0xbdc8 EventHandler::OnEvent(char const*, char const*) + 21 (FlutterIrisEventHandler.mm:21)
5  AgoraRtcWrapper                0x65714 agora::iris::rtc::RtcEngineEventHandler::onLocalAudioStateChanged(agora::rtc::LOCAL_AUDIO_STREAM_STATE, agora::rtc::LOCAL_AUDIO_STREAM_ERROR) + 1459 (string:1459)
6  AgoraRtcKit                    0x3d8c88 (누락 UUID 1ec6daa1d1f23e3dae9a2685ee010cd3)
7  AgoraRtcKit                    0x3d6fac (누락 UUID 1ec6daa1d1f23e3dae9a2685ee010cd3)
8  AgoraRtcKit                    0x36b42c (누락 UUID 1ec6daa1d1f23e3dae9a2685ee010cd3)
9  AgoraRtcKit                    0x36b39c (누락 UUID 1ec6daa1d1f23e3dae9a2685ee010cd3)
10 AgoraCore                      0xc9d0 (누락 UUID e9a448df109039e8a72e96e8aca0bd93)
11 AgoraCore                      0xdae0 (누락 UUID e9a448df109039e8a72e96e8aca0bd93)
12 AgoraCore                      0xd6b4 (누락 UUID e9a448df109039e8a72e96e8aca0bd93)
13 AgoraCore                      0xcc78 (누락 UUID e9a448df109039e8a72e96e8aca0bd93)
14 AgoraCore                      0x18358 (누락 UUID e9a448df109039e8a72e96e8aca0bd93)
15 libsystem_pthread.dylib        0x19ac _pthread_start
16 libsystem_pthread.dylib        0xe68 thread_start
```

```
Crashed: AgoraEvent
0  FirebaseCrashlytics            0x223ac FIRCLSProcessRecordAllThreads + 393 (FIRCLSProcess.c:393)
1  FirebaseCrashlytics            0x2278c FIRCLSProcessRecordAllThreads + 424 (FIRCLSProcess.c:424)
2  FirebaseCrashlytics            0x187b4 FIRCLSHandler + 34 (FIRCLSHandler.m:34)
3  FirebaseCrashlytics            0x129b0 __FIRCLSExceptionRecord_block_invoke + 232 (FIRCLSException.mm:232)
4  libdispatch.dylib              0x3a2c _dispatch_client_callout + 20
5  libdispatch.dylib              0x12c90 _dispatch_lane_barrier_sync_invoke_and_complete + 56
6  FirebaseCrashlytics            0x11790 FIRCLSExceptionRecord + 234 (FIRCLSException.mm:234)
7  FirebaseCrashlytics            0x12458 FIRCLSExceptionRecordNSException + 126 (FIRCLSException.mm:126)
8  FirebaseCrashlytics            0x11390 FIRCLSTerminateHandler() + 398 (FIRCLSException.mm:398)
9  libc++abi.dylib                0x10f18 std::__terminate(void (*)()) + 20
10 libc++abi.dylib                0x13c1c __cxa_get_exception_ptr + 34
11 libc++abi.dylib                0x13bc8 __cxxabiv1::exception_cleanup_func(_Unwind_Reason_Code, _Unwind_Exception*) + 138
12 libobjc.A.dylib                0x1689c objc_exception_throw + 420
13 Foundation                     0x1311f0 _userInfoForFileAndLine + 202
14 Flutter                        0x12d54 (누락 UUID b5f419596300357c85f161f5e6e0920d)
15 agora_rtc_engine               0xbdc8 EventHandler::OnEvent(char const*, char const*) + 21 (FlutterIrisEventHandler.mm:21)
16 AgoraRtcWrapper                0x65714 agora::iris::rtc::RtcEngineEventHandler::onLocalAudioStateChanged(agora::rtc::LOCAL_AUDIO_STREAM_STATE, agora::rtc::LOCAL_AUDIO_STREAM_ERROR) + 1459 (string:1459)
17 AgoraRtcKit                    0x3d8c88 agora::rtc::RtcEngineParameters::setInEarMonitoringVolume(int) + 57016
18 AgoraRtcKit                    0x3d6fac agora::rtc::RtcEngineParameters::setInEarMonitoringVolume(int) + 49628
19 AgoraRtcKit                    0x36b42c (누락 UUID 1ec6daa1d1f23e3dae9a2685ee010cd3)
20 AgoraRtcKit                    0x36b39c (누락 UUID 1ec6daa1d1f23e3dae9a2685ee010cd3)
21 AgoraCore                      0xc9d0 ahpl_perf_set_callback + 1500
22 AgoraCore                      0xdae0 ahpl_mpq_itc_ack + 340
23 AgoraCore                      0xd6b4 ahpl_mpq_loop + 84
24 AgoraCore                      0xcc78 ahpl_perf_set_callback + 2180
25 AgoraCore                      0x18358 ahpl_profile_clear_registered + 2488
26 libsystem_pthread.dylib        0x19ac _pthread_start + 148
27 libsystem_pthread.dylib        0xe68 thread_start + 8
```

**Comment by nizwar** [comment]
Hi dev, i think i know how to produce this crash... i got crash once i immediately minimize the application during agora join progress, idk it might be once agora checking userJoined or something because for sure its happened after joinChannel called success

as you can see on this image,  i log user's behavior for app life state and it getting crashes once app state was inactivated 

![1A2A6030-44C9-4029-9CBA-AD2555F62852](https://user-images.githubusercontent.com/33113137/183215066-df683983-e814-4eb1-9781-96e2a2bb43d8.png)

**Comment by littleGnAl** [comment]
Thanks for the more detail @nizwar @Hwan-seok , I finally found the root cause of this issue, this crash is caused by the `RtcEngine.destroy` is not be called expectedly, say that directly kill the APP. I will release a hot fix next week to fix this it, sorry for this issue blocking your guys for so long.

**Comment by littleGnAl** [comment]
Relative https://github.com/deven98/shake/issues/12

**Comment by littleGnAl** [comment]
Sorry for the late response, this issue had been fixed and merged into the master branch, but it will be released to the pub in the next major version due to another release schedule internally, sorry about that.

**Comment by tselmeggkh** [comment]
Has it been released to the pub yet?

**Comment by littleGnAl** [comment]
This fix has been released `agora_rtc_engine: ^5.3.1`.
https://pub.dev/packages/agora_rtc_engine

**Comment by fisforfaheem** [comment]
agora needs to fix video quality
